content/blog/2020/07/2020-07-30-github-checks-api-plugin-coding-phase-2.adoc

```diff
@@ -0,0 +1,149 @@
+---
+layout: post
+title: "GitHub Checks API Plugin Project - Coding Phase 2"
+tags:
+- github
+- api
+- plugins
+- developer
+- gsoc
+- gsoc2020
+author: XiongKezhi
+description: "Google Summer of Code 2020 - GitHub Checks API Plugin project update for Coding Phase 2"
+opengraph:
+  image: /images/gsoc/opengraph.png
+---
+
+Another great coding phase for link:https://www.jenkins.io/projects/gsoc/2020/projects/github-checks/[GitHub Checks API Project] ends!
+In this phase, we focused on consuming the checks API in two widely used plugin:
+
+* link:https://plugins.jenkins.io/warnings-ng/[Warnings NG Plugin] 
+* link:https://plugins.jenkins.io/code-coverage-api/[Code Coverage API Plugin]
+
+Besides the external usage, we have also split the general checks API from its GitHub implementation and released both of them:
+
+* link:https://plugins.jenkins.io/checks-api/[Checks API Plugin]
+* link:https://plugins.jenkins.io/github-checks/[GitHub Checks Plugin]
+
+.Coding Phase 2 Demo [starts from 25:20]
+video::b67I6spBdTg[youtube,width=800,height=420]
+
+== Warning Checks
+
+The newly released https://github.com/jenkinsci/warnings-ng-plugin/releases/tag/warnings-ng-8.4.0[Warnings NG plugin 8.4.0] will use checks API to publish different check runs for different static analysis tools.
+Without leaving GitHub, users are now able to see the analysis report they interested in.
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/warning-checks.png[Warning Checks Summary]
+
+On GitHub's conversation tab for each PR, users will see summaries for those checks like the screenshot above. The summaries will include:
+
+* The status that indicates the quality gate
+* The name of the analysis tool used
+* A short message that indicates statistics of new and total issues
+
+More fine-grained statistics can be found in the `Details` page.
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/severity-statistics.png[Severity Statis]
+
+Another practical feature is the annotation for specific lines of code. Users can now review the code alone with the annotations.
+
+image::/images/post-images/2020/07-github-checks-api-plugin-coding-phase-2/annotations.png[Warning Annotations]
+
+=== Try It
+
+In https://github.com/jenkinsci/warnings-ng-plugin/releases/tag/warnings-ng-8.4.0[Wanings NG plugin 8.4.0], the warning checks is set as a default feature. 
+Therefore, you can get those checks for your own GitHub project just in a few steps:
+
+1. Update Warnings NG plugin to 8.4.0
+2. Install link:https://plugins.jenkins.io/checks-api/[Checks API plugin] and link:https://plugins.jenkins.io/github-checks/[GitHub Checks plugin] on your Jenkins instance
```

Comment:
The checks api plugin is a dependency of the warnings plugin. So only the GitHub-Checks Plugin needs to be selected manually.